main.tf

```diff
@@ -15,25 +15,42 @@ resource "azurerm_monitor_action_group" "eventpipeline" {
   }
 }
 
-resource "azurerm_monitor_scheduled_query_rules_alert" "this" {
-  for_each = local.rules
+resource "azurerm_monitor_scheduled_query_rules_alert_v2" "this" {
+  for_each            = local.rules
   name                = each.key
   location            = var.log_analytics_workspace.location
   resource_group_name = var.log_analytics_workspace.resource_group_name
   tags                = var.tags
+
   action {
-    action_group = local.action_group
+    action_groups = local.action_group
   }
-  data_source_id = var.log_analytics_workspace.id
-  description    = each.value.description
-  enabled        = true
-  # Count all requests with server error result code grouped into 5-minute bins
-  query       = file(each.value.query_path)
-  severity    = 1
-  frequency   = 5
-  time_window = each.value.time_window
-  trigger {
-    operator  = "GreaterThan"
-    threshold = 0
+
+  scopes                = [var.log_analytics_workspace.id]
+  description           = each.value.description
+  enabled               = true
+  severity              = 0
+  skip_query_validation = true
+
+  evaluation_frequency = "PT5M"
```

Comment:
I think it is reasonable to allow to set the frequency, as you can set the duration.